Sources/SwiftParser/Declarations.swift

```diff
@@ -1697,12 +1697,91 @@ extension Parser {
   ///     postfix-operator-declaration → 'postfix' 'operator' operator
   ///     infix-operator-declaration → 'infix' 'operator' operator infix-operator-group?
   ///     infix-operator-group → ':' precedence-group-name
+
+  struct OperatorDeclIntroducer {
+    var unexpectedBeforeFixity: RawUnexpectedNodesSyntax?
+    var fixity: RawTokenSyntax
+    var unexpectedBeforeOperatorKeyword: RawUnexpectedNodesSyntax?
+    var operatorKeyword: RawTokenSyntax
+  }
+
+  mutating func parseOperatorDeclIntroducer(_ attrs: DeclAttributes, _ handle: RecoveryConsumptionHandle) -> OperatorDeclIntroducer {
+
+    let fixityText = [Keyword.prefix, .infix, .postfix].map({ $0.defaultText })
+
+    var unexpectedBeforeFixity = RawUnexpectedNodesSyntax(attrs.attributes?.elements ?? [], arena: self.arena)
+
+    var fixity = self.missingToken(.prefix)
+    var leftModifiers: [RawDeclModifierSyntax?] = []
+
+    if let modifiers = attrs.modifiers?.elements {
+      for modifier in modifiers {
+        if fixity.presence == .missing,
+          fixityText.contains(modifier.name.tokenText)
+        {
+          fixity = modifier.name
+          break
+        }
+        leftModifiers.append(modifier)
+      }
+    }
+
+    var (unexpectedBeforeOperatorKeyword, operatorKeyword) = self.expect(.keyword(.operator))
+
+    // Sometimes we could not parse the fixity into attr.modifiers
+    // so we need extract it from unexpectedBeforeOperatorKeyword manually
+    if fixity.presence == .missing,
+      let elements = unexpectedBeforeOperatorKeyword?.elements
+    {
+
+      var unexpectedBeforeFixityElements: [RawSyntax?] = []
+      var unexpectedBeforeOperatorElements: [RawSyntax?] = []
+      for element in elements {
+        if fixity.presence == .missing,
+          let token = element.as(RawTokenSyntax.self),
+          fixityText.contains(token.tokenText)
+        {
+          fixity = token
+          continue
+        }
+
+        if fixity.presence == .missing {
+          unexpectedBeforeFixityElements.append(element)
+        } else {
+          unexpectedBeforeOperatorElements.append(element)
+        }
+      }
+
+      if !unexpectedBeforeFixityElements.isEmpty {
+        unexpectedBeforeFixity = RawUnexpectedNodesSyntax(combining: unexpectedBeforeFixity, RawUnexpectedNodesSyntax(unexpectedBeforeFixityElements, arena: self.arena), arena: self.arena)
+      }
+
+      if !unexpectedBeforeOperatorElements.isEmpty {
+        unexpectedBeforeOperatorKeyword = RawUnexpectedNodesSyntax(unexpectedBeforeOperatorElements, arena: self.arena)
+      } else {
+        unexpectedBeforeOperatorKeyword = nil
+      }
+
+    }
```

Comment:
To produce better diagnostic when fixity could not be parsed into `attrs.modifiers`, e.g. 
```swift
; // ';' separator here has higher precedence and prefix is a contextual keyword
prefix operator ++
```
I added these codes to extract the fixity from `unexpectedBeforeOperatorKeyword`, which looks IMO awful. Do you have better idea about this?